notebooks/generative-ai/chatbot.ipynb

```diff
@@ -0,0 +1,433 @@
+     ]
+    }
+   ],
+   "source": [
+    "from elasticsearch import Elasticsearch\n",
+    "\n",
+    "elasticsearch_client = Elasticsearch(\n",
+    "    cloud_id=ELASTIC_CLOUD_ID,\n",
+    "    basic_auth=(ELASTIC_USERNAME, ELASTIC_PASSWORD)\n",
+    ")\n",
+    "\n",
+    "print('Connected to Elasticsearch\\n', elasticsearch_client.info())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create index 🗄️\n",
+    "\n",
+    "We'll create an Elasticsearch index to store documents along with the generated vector embeddings. This will allow us to execute vector search when retrieving documents for our query.\n",
+    "\n",
+    "Since we're using OpenAI's `text-embedding-ada-002` model, we need a 1536-dimensional [dense_vector](https://www.elastic.co/guide/en/elasticsearch/reference/current/dense-vector.html) field to store the embeddings."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mappings = {\n",
+    "    \"properties\": {\n",
+    "        \"text\": { \"type\": \"keyword\" },\n",
+    "        \"vector\": {\n",
+    "            \"type\": \"dense_vector\",\n",
+    "            \"dims\": 1536,\n",
+    "            \"index\": True,\n",
+    "            \"similarity\": \"cosine\"\n",
+    "        }\n",
+    "    }\n",
+    "}\n",
+    "\n",
+    "elasticsearch_client.indices.create(\n",
+    "    index='workplace-docs',\n",
+    "    mappings=mappings\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Load and process documents 📄\n",
+    "\n",
+    "Time to load some data! We'll be using the workplace search example data, which is a list of employee documents and policies."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 74,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Successfully loaded 15 documents\n"
+     ]
+    }
+   ],
+   "source": [
+    "import json\n",
+    "from urllib.request import urlopen\n",
+    "\n",
+    "url = \"https://raw.githubusercontent.com/elastic/elasticsearch-labs/main/example-apps/workplace-search/example-data/data.json\"\n",
+    "\n",
+    "response = urlopen(url)\n",
+    "\n",
+    "workplace_docs = json.loads(response.read())\n",
+    "\n",
+    "print(f'Successfully loaded {len(workplace_docs)} documents')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As we're chatting with our bot, it will run semantic searches on the index to find the relevant documents. In order for this to be accurate, we need to split the full documents into small chunks (also called passages). This way the semantic search will find the passage within a document that most likely answers our question.\n",
```

Comment:
nit: I'd suggest making the _splitting/chunking_ step its own sub-heading, so you'd have two subheadings something like:

- Load ~and process~ documents
- Chunk documents into passages